cpp/examples/ad_odeint_example.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Ralf Hannemann-Tamas
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "ad/ad.hpp"
+
+#include "memilio/utils/compiler_diagnostics.h"
+
+/* disable warnings of odint.hpp */
+
+GCC_CLANG_DIAGNOSTIC(push)
+GCC_CLANG_DIAGNOSTIC(ignored "-Wdeprecated-declarations")
+
+#include "boost/numeric/odeint.hpp"
+
+GCC_CLANG_DIAGNOSTIC(pop)
+
+/**
+ * This program shows that  boost::numeric::odeint::runge_kutta_cash_karp54 can be fully
+ * algorithmically diffentiated using the algorithmic differentiation (AD) data types of ad/ad.hpp 
+ */
```

Comment:
```suggestion
// This program shows that  boost::numeric::odeint::runge_kutta_cash_karp54 can be fully
// algorithmically diffentiated using the algorithmic differentiation (AD) data types of ad/ad.hpp 
```
Do not use doc comments (note the double star) arbitrarily. For strictly in-code documentation, we prefer single line comments over C-style comments.

I will not mark the other comments in this file, but they should be changed as well.